app/models/wiki_content.rb

```diff
@@ -68,8 +68,7 @@ def attachments
   # Returns the mail adresses of users that should be notified
   def recipients
     notified = project.notified_users
-    notified.reject! { |user| !visible?(user) }
-    notified.map(&:mail)
+    notified.reject { |user| !visible?(user) }
```

Comment:
If you like, you could change this to `select(&:visible?)` (instead of "reject not").